src/XRP/core-xrpl-client.ts

```diff
@@ -15,11 +15,15 @@ import { GetFeeResponse } from './Generated/web/org/xrpl/rpc/v1/get_fee_pb'
 import TransactionStatus from './transaction-status'
 import RawTransactionStatus from './raw-transaction-status'
 import { XrpNetworkClient } from './xrp-network-client'
-import XrpError from './xrp-error'
+import XrpError, { XrpErrorType } from './xrp-error'
 import { LedgerSpecifier } from './Generated/web/org/xrpl/rpc/v1/ledger_pb'
 import TransactionResult from './model/transaction-result'
 import isNode from '../Common/utils'
 
+async function sleep(milliseconds: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds))
+}
+
```

Comment:
It could - in this case the `CoreXrplClientInterface` was a way to create a `FakeCoreXrplClient` such that we only had to implement a few fake methods on it (the ones needed specifically by `ReliableSubmissionXrpClient`), and then could type the instance of `CoreXrplClient` as a `CoreXrpClientInterface` in `ReliableSubmissionXrpClient` and have it accept the fake.  We probably should make the actual underlying class used in production officially conform to the purported type in `ReliableSub`.